pkgs/development/libraries/tbb/2021_8.nix

```diff
@@ -0,0 +1,58 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, fetchurl
+, cmake
+}:
+
+let
+  common = import ./common.nix { inherit lib; };
+in
+stdenv.mkDerivation rec {
+  pname = "tbb";
+  version = "2021.8.0";
+
+  outputs = [ "out" "dev" ];
+
+  src = fetchFromGitHub {
+    owner = "oneapi-src";
+    repo = "oneTBB";
+    rev = "v${version}";
+    hash = "sha256-7MjUdPB1GsPt7ZkYj7DCisq20X8psljsVCjDpCSTYT4=";
+  };
+
+  nativeBuildInputs = [
+    cmake
+  ];
+
+  patches = [
+    # Fix musl build; vendored from https://github.com/oneapi-src/oneTBB/pull/899
+    ./musl.patch
+
+    # Fix/suppress warnings on gcc12.1 from https://github.com/oneapi-src/oneTBB/pull/866
+    (fetchurl {
+      url = "https://patch-diff.githubusercontent.com/raw/oneapi-src/oneTBB/pull/866.patch";
+      sha256 = "sha256-zT+tRPnPVGUqKvob0RWjASRm4bPYYLtaaqTzA4EdU/8=";
+    })
+  ];
+
+
+  # Fix build with modern gcc
+  # In member function 'void std::__atomic_base<_IntTp>::store(__int_type, std::memory_order) [with _ITp = bool]',
+
+  # Disable failing test on musl
+  # test/conformance/conformance_resumable_tasks.cpp:37:24: error: ‘suspend’ is not a member of ‘tbb::v1::task’; did you mean ‘tbb::detail::r1::suspend’?
+  postPatch = ''
+    substituteInPlace cmake/compilers/GNU.cmake \
+        --replace '-Wfatal-errors)' '-Wfatal-errors -Wno-error=stringop-overflow)'
+  '' + lib.optionalString stdenv.hostPlatform.isMusl ''
+    substituteInPlace test/CMakeLists.txt \
+      --replace 'conformance_resumable_tasks' ""
+  '';
+
+  meta = common.meta // {
+    # build with gcc fails on i686-linux (32-bit)
+    # https://github.com/oneapi-src/oneTBB/issues/1030
+    broken = stdenv.cc.isGNU && stdenv.isx86_32;
```

Comment:
Oh, it's a compiler bug. Those are usually not too hard to workaround once understood. I'll try to reproduce locally.

Reply:
https://gcc.gnu.org/PR108854 confirmed it's a code generation bug related to `-O3` and instruction selection. I suggest the following workaround until `gcc` is fixed:

```diff
--- a/pkgs/development/libraries/tbb/2021_8.nix
+++ b/pkgs/development/libraries/tbb/2021_8.nix
@@ -50,6 +50,10 @@ stdenv.mkDerivation rec {
       --replace 'conformance_resumable_tasks' ""
   '';

+  # Workaround gcc-12 ICE when using -O3:
+  #   https://gcc.gnu.org/PR108854
+  NIX_CFLAGS_COMPILE = lib.optionals (stdenv.cc.isGNU && stdenv.isx86_32) [ "-O2" ];
+
   meta = common.meta // {
     # build with gcc fails on i686-linux (32-bit)
     # https://github.com/oneapi-src/oneTBB/issues/1030
```

Reply:
With this the build succeeds. Awesome!